File: 1635-NumberOfGoodPairs/1635-NumberOfGoodPairs.java
// Last updated: 26/12/2025, 15:59:30
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int totalPairs=0;
        
        for(int i=0; i<nums.length; i++){
           for(int j=0; j<nums.length; j++){
               if(i!=j){
                   if(nums[i]==nums[j]){
                       totalPairs++;
                   }
               }
           }
            }
        return totalPairs/2;
        }
        
    
}

File: 1635-NumberOfGoodPairs/1635-NumberOfGoodPairs.js
// Last updated: 26/12/2025, 15:59:27
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    let map = new Map();
    let result = 0;

    for(let n of nums){
        map.set(n, (map.get(n) || 0)+1)
    }
    

    for(let [k,v] of map.entries()){
        if(v > 1){
            result += maxPossbilities(v);
        }
        
    }

    return result;
};

var maxPossbilities = function(nums){
    return (nums*(nums-1))/2
}
